File: name-generator/python/name_generator.py
#!/usr/bin/python3.7

import random
import os


def parse_name_file(file_name: str) -> list:
    with open(file_name) as name_file:
        parsed_names = []
        split_names = name_file.read().split()
        for i in range(0, len(split_names), 4):
            parsed_names.append(split_names[i])
    return parsed_names


def get_random_name(gender):
    first_name = ''
    if gender == 'male':
        first_name = random.choice(male_names)
    elif gender == 'female':
        first_name = random.choice(female_names)
    else:
        print('Invalid gender')
    last_name = random.choice(last_names)
    print(
        f'Random name: {first_name[0]}{first_name[1:].lower()} {last_name[0]}{last_name[1:].lower()}'
    )


last_names = parse_name_file(os.path.abspath(
    'name-generator/name_data/dist.all.last'))
female_names = parse_name_file(
    os.path.abspath('name-generator/name_data/dist.female.first'))
male_names = parse_name_file(os.path.abspath(
    'name-generator/name_data/dist.male.first'))


get_random_name('male')
get_random_name('female')

File: name-generator/java/NameGenerator.java
import java.io.File;
import java.io.BufferedReader;
import java.io.FileReader;
import java.util.ArrayList;
import java.util.Random;;

public class NameGenerator {

    static ArrayList<String> lastNames = readFile("../name_data/dist.all.last");
    static ArrayList<String> femaleNames = readFile("../name_data/dist.female.first");
    static ArrayList<String> maleNames = readFile("../name_data/dist.male.first");
    static Random random = new Random();

    public static void main(String[] args) {
        String firstName = getRandomFirstName("female");
        String lastName = getRandomLastName();
        String firstLetterFirstName = firstName.substring(0, 1);
        String restFirstName = firstName.substring(1);
        String firstLetterLastName = lastName.substring(0, 1);
        String restLastName = lastName.substring(1);

        System.out.println("Random name: " + firstLetterFirstName + restFirstName.toLowerCase() + " "
                + firstLetterLastName + restLastName.toLowerCase());
    }

    public static ArrayList<String> readFile(String filePath) {
        File file = new File(filePath);
        BufferedReader reader;
        try {
            String line;
            ArrayList<String> nameList = new ArrayList<String>();
            reader = new BufferedReader(new FileReader(file.getAbsolutePath()));
            while ((line = reader.readLine()) != null) {
                String[] splitString = line.split("( )+");
                String name = splitString[0];
                nameList.add(name);
            }
            reader.close();
            return nameList;
        } catch (Exception e) {
            System.out.println(e.getMessage());
            return new ArrayList<String>();
        }
    }

    public static String getRandomFirstName(String gender) {
        if (gender == "male") {
            return maleNames.get(random.nextInt(maleNames.size()));
        } else if (gender == "female") {
            return femaleNames.get(random.nextInt(femaleNames.size()));
        } else {
            return "";
        }
    }

    public static String getRandomLastName() {
        return lastNames.get(random.nextInt(lastNames.size()));
    }
}